lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflectDatumWithAnonymousEnum.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.avro.reflect;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * https://issues.apache.org/jira/browse/AVRO-1851
+ */
+public class TestReflectDatumWithAnonymousEnum {
+  private static Pojo pojo;
+
+  @BeforeClass
+  public static void init() {
+    pojo = new Pojo();
+    Person person = new Person();
+    person.setAddress("Address");
+    pojo.setTestEnum(TestEnum.V);
+    pojo.setPerson(person);
+  }
+
+  // Properly serializes and deserializes a POJO with an enum instance (TestEnum#V)
+  @Test
+  public void handleProperlyEnumInstances() throws IOException {
+    byte[] output = serialize(pojo);
+    Pojo deserializedPojo = deserialize(output);
+    assertEquals(pojo, deserializedPojo);
+  }
+
+  // The test fails because the Schema doesn't support null value for the Person's name
+  @Test(expected = NullPointerException.class)
+  public void avroEnumWithNotNullTest() throws IOException {
```

Comment:
I am not sure this test is needed here. Most probably this functionality is already covered in another test. It has nothing to do with "anonymous enums". It just checks that a non-nullable field (Person#name) cannot have a null value with non-union schema.

Reply:
I agree with you on this.

Reply:
Removed it!
And made the test more fancy by using anonymous instances for Pojo and Person too.